File: gin/src/response/response.go
package response

import "github.com/gin-gonic/gin"

// Response mengirimkan response JSON dengan status, data, pesan, dan metadata ke client.
func Response(statusCode int, data interface{}, message string, c *gin.Context) {
	c.JSON(statusCode, gin.H{
		"payload": data,
		"message": message,
		"metadata": gin.H{
			"prev":    "",
			"next":    "",
			"current": "",
		},
	})
}

File: express/src/response/response.js
// Fungsi untuk mengirimkan respons JSON dengan status code, data, pesan, dan metadata
const response = (statusCode, data, message, res) => {
  res.status(statusCode).json({
    payload: data,
    message,
    metadata: {
      prev: "",
      next: "",
      current: "",
    },
  });
};

export { response };
